scripts/tests/test_zkai_d128_gate_value_compact_preprocessed_gate.py

```
@@ -0,0 +1,264 @@
+        summary["rows"][1]["local_binary_accounting"]["stwo_grouped_breakdown"]["fixed_overhead"] += 1
+        with self.assertRaisesRegex(gate.GateValueCompactGateError, "baseline stwo_grouped_breakdown drift"):
+            build_payload(summary, include_mutations=False)
+
+    def test_rejects_duplicate_accounting_row(self):
+        summary = cli_summary()
+        summary["rows"].append(copy.deepcopy(summary["rows"][0]))
+        with self.assertRaisesRegex(gate.GateValueCompactGateError, "duplicate accounting row"):
+            build_payload(summary, include_mutations=False)
+
+    def test_rejects_envelope_size_drift(self):
+        with self.assertRaisesRegex(gate.GateValueCompactGateError, "compact envelope JSON size drift"):
```

Ruff (0.15.12)

[warning] Use `pytest.raises` instead of unittest-style `assertRaisesRegex`

Replace `assertRaisesRegex` with `pytest.raises`

(PT027)